### PR TITLE
psst: 2022-10-13 -> 2023-05-13

### PR DESCRIPTION
**Pull request by ambroisie**
###### Description of changes

I wanted to update the program to take advantage of https://github.com/jpochyla/psst/pull/392.

Changelog: https://github.com/jpochyla/psst/compare/d70ed8104533dc15bc36b989ba8428872c9b578f...f94af14aa9fdd3d59cd92849fa7f076103b37a70

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/psst/update.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+set -euo pipefail
+
+if [ -z "$rev" ]; then
+    rev="$(wget -O- "${TOKEN_ARGS[@]}" "https://api.github.com/repos/jpochyla/psst/commits?per_page=1" | jq -r '.[0].sha')"
+fi
+
+version="unstable-$(date +%F)"
+
+# Woodpecker repository
```
Comment (ambroisie) [inline pkgs/applications/audio/psst/update.sh]:
```suggestion
# Sources
```

**Comment by vbrandl** [comment]
Just trying to understand the changes:

* `make-build-reproducible.patch` updates the `build.rs` to use hardcoded variables for the git rev, build time and repo URL.
* `update.sh` uses the Github API to obtain the latest commit rev and updates the nix expression and `make-build-reproducible.patch` to reflect the current date and commit rev.
* I'm not quite sure what the `passthru` is doing. Does this automatically run the `update.sh` script to check for changes?
* Is it possible to schedule a daily/weekly run of the `update.sh` script and create automated PRs if something changed?

**Comment by ambroisie** [comment]
> make-build-reproducible.patch updates the build.rs to use hardcoded variables for the git rev, build time and repo URL.

Correct.

> update.sh uses the Github API to obtain the latest commit rev and updates the nix expression and make-build-reproducible.patch to reflect the current date and commit rev.

Correct.

> I'm not quite sure what the passthru is doing. Does this automatically run the update.sh script to check for changes?

The point of `passthru.updateScript` is [detailed in the docs](https://nixos.org/manual/nixpkgs/unstable/#var-passthru-updateScript). TL;DR: this allows the maintainer and/or @r-ryantm to update the package.

> Is it possible to schedule a daily/weekly run of the update.sh script and create automated PRs if something changed?

This is what @r-ryantm does. However it is unlikely to work for this package due to the `cargoLock.outputHashes` which need manual updating, and aren't easily scriptable in `update.sh` (that I know of).

**Review by vbrandl — APPROVED** [review]
LGTM

**Comment by peterhoeg** [comment]
hardcoding the commit in the patch file is not great

**Pull request merged**